packages/webapp/pages/onboarding.tsx

```diff
@@ -0,0 +1,223 @@
+import React, { ReactElement, useEffect, useRef, useState } from 'react';
+import { useOnboardingContext } from '@dailydotdev/shared/src/contexts/OnboardingContext';
+import { useFeaturesContext } from '@dailydotdev/shared/src/contexts/FeaturesContext';
+import { ProgressBar } from '@dailydotdev/shared/src/components/fields/ProgressBar';
+import Logo, { LogoPosition } from '@dailydotdev/shared/src/components/Logo';
+import classNames from 'classnames';
+import ConditionalWrapper from '@dailydotdev/shared/src/components/ConditionalWrapper';
+import { IntroductionOnboardingTitle } from '@dailydotdev/shared/src/components/onboarding/IntroductionOnboarding';
+import AuthOptions from '@dailydotdev/shared/src/components/auth/AuthOptions';
+import { AuthTriggers } from '@dailydotdev/shared/src/lib/auth';
+import { FilterOnboarding } from '@dailydotdev/shared/src/components/onboarding';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import { MemberAlready } from '@dailydotdev/shared/src/components/onboarding/MemberAlready';
+import {
+  OnboardingFilteringTitle,
+  OnboardingV2,
+} from '@dailydotdev/shared/src/lib/featureValues';
+import classed from '@dailydotdev/shared/src/lib/classed';
+import { useRouter } from 'next/router';
+import { useAnalyticsContext } from '@dailydotdev/shared/src/contexts/AnalyticsContext';
+import {
+  AnalyticsEvent,
+  TargetType,
+} from '@dailydotdev/shared/src/lib/analytics';
+import { OnboardingStep } from '@dailydotdev/shared/src/components/onboarding/common';
+import { OnboardingMode } from '@dailydotdev/shared/src/graphql/feed';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+import { Loader } from '@dailydotdev/shared/src/components/Loader';
+import { NextSeo, NextSeoProps } from 'next-seo';
+import { defaultOpenGraph, defaultSeo } from '../next-seo';
+
+const versionToTitle: Record<OnboardingFilteringTitle, string> = {
+  [OnboardingFilteringTitle.Control]: 'Choose topics to follow',
+  [OnboardingFilteringTitle.V1]: 'What topic best describes you?',
+  [OnboardingFilteringTitle.V2]: 'Which topics resonate with you the most?',
+  [OnboardingFilteringTitle.V3]: `Pick the topics you'd love to dive into`,
+  [OnboardingFilteringTitle.V4]: 'Choose the topics you’re passionate about',
+};
+
+const Title = classed('h2', 'font-bold typo-title2');
+
+interface AuthProps {
+  isAuthenticating: boolean;
+  isLoginFlow: boolean;
+}
+
+const maxAuthWidth = 'tablet:max-w-[30rem]';
+
+const Container = classed(
+  'div',
+  'flex flex-col overflow-x-hidden items-center min-h-[100vh] w-full h-full max-h-[100vh] flex-1 z-[100] bg-theme-bg-primary',
+);
+
+const seo: NextSeoProps = {
+  title: 'Get started',
+  openGraph: { ...defaultOpenGraph },
+  ...defaultSeo,
+};
+
+export function OnboardPage(): ReactElement {
+  const router = useRouter();
+  const [isTracked, setIsTracked] = useState(false);
```

Comment:
`useRef` should be better to avoid causing re-renders.

Reply:
Agreed.